pkg/receiver/mysqlreceiver/client.go

```diff
@@ -0,0 +1,135 @@
+package mysqlreceiver 
+
+import (
+	"database/sql"
+	"fmt"
+	"log"
+	"encoding/json"
+	"github.com/go-sql-driver/mysql"
+)
+
+type client interface {
+	Connect() error
+	getRecords() (map[int]string, error)
+	Close() error
+}
+
+type mySQLClient struct {
+	connStr string
+	client  *sql.DB
+}
+
+var _ client = (*mySQLClient)(nil)
+
+func newMySQLClient(conf *Config) client {
+	driverConf := mysql.Config{
+		User:                 conf.Username,
+		Passwd:               conf.Password,
+		Net:                  conf.Transport,
+		Addr:                 conf.Endpoint,
+		DBName:               conf.Database,
+		AllowNativePasswords: conf.AllowNativePasswords,
+	}
+	connStr := driverConf.FormatDSN()
+
+	return &mySQLClient{
+		connStr: connStr,
+	}
+}
+
+func (c *mySQLClient) Connect() error {
+	clientDB, err := sql.Open("mysql", c.connStr)
+	if err != nil {
+		return fmt.Errorf("unable to connect to database: %w", err)
+	}
+	c.client = clientDB
+	return nil
+}
+
+// getInnodbStats queries the db for innodb metrics.
+func (c *mySQLClient) getRecords() (map[int]string,error) {
+	query := "select * from Persons;"
+	return Query(*c, query)
+}
+
+func Query(c mySQLClient, query string) (map[int]string,error) {
+	rows, err := c.client.Query(query)
+	if err != nil {
+		log.Printf("error in executing sql query")
```

Comment:
Please use [go.uber.org/zap](https://pkg.go.dev/go.uber.org/zap) for logging, I see that it is commonly used in OTC, you can find example usage in other receivers